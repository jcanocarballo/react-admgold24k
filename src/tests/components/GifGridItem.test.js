import {shallow} from 'enzyme'
const { GifGridItem } = require("../../components/GifGridItem");

describe('Pruebas del componente GifGridItem', () =>{
  
  const title = 'Un titulo';
  const url = 'https://localhost/algo.jpg'
  
  let wraper = shallow(<GifGridItem title={title} url={url}></GifGridItem>);
  beforeEach( () => {
    wraper = shallow(<GifGridItem title={title} url={url}></GifGridItem>);
  });

  test('debe mostrar el componente <GifGridItem /> correctamente', () => {
    expect(wraper).toMatchSnapshot();  
  });

  test('debe de tener un parrafo con el titulo', () => {
    const p = wraper.find('p').text();
    expect(p.trim()).toBe(title);
  });

  test('debe tener la imagen igual al url y alt de los props', () => {
    const img = wraper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('debe de tener la clase animate__fadeIn ', () => {
    const div = wraper.find('div');
    const clase = div.prop('className');
    expect(clase.includes('animate__fadeIn')).toBe(true);
  });
  
});