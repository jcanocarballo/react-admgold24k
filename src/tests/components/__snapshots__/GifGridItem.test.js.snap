// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGridItem debe mostrar el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
